src/db_adapters/blocks.rs

```diff
@@ -39,3 +40,25 @@ pub(crate) async fn store_block(
         }
     }
 }
+
+/// Gets the latest block's height from database
+pub(crate) async fn latest_block_height(
+    pool: &actix_diesel::Database<PgConnection>,
+) -> Result<u64, String> {
+    let result: Vec<models::blocks::Block> = schema::blocks::table
+        .limit(1)
+        .order(schema::blocks::dsl::block_height.desc())
+        .load_async(&pool)
+        .await
+        .unwrap_or_else(|_| vec![]);
+
+    if result.is_empty() {
+        return Err("No blocks found in database".to_string());
+    }
+
+    Ok(result[0]
+        .block_height
+        .clone()
+        .to_u64()
+        .expect("Failed to convert block_height from BigDecimal to u64"))
```

Comment:
With the following things it could be easier to handle (btw, it might be useful to return `Option<u64>` or even `Option<near_primitives::BlockHeight>`

```suggestion
        .select((schema::blocks::columns::BlockHeight))
        .first_async(&pool)
        .await
        .optional()
        .map_err(|err| format!("DB Error: {}", err))?
        .map(|(block_height,)| block_height.to_u64())
```